### PR TITLE
refactor: remove a lot of useless api calls by fixing useEffect calls

### PR DESCRIPTION
**Pull request by Gamer92000**

### CODE REVIEW

File: src/pages/manage/users/activeTrainerTable.tsx [hunk 1]
```diff
@@ -16,64 +40,37 @@ interface User {
...
   const redraw = () => draw({});
 
-  const deleteTrainer = async (id: string) => {
-    api.execute(Routes.deleteTrainer({ trainerId: id })).then(() => {
-      message.success(t(Translations.userManagement.trainerDeleted));
-      setLoading(true);
+  const [isMounted, setIsMounted] = useState(true);
+
+  const loadData = () => {
+    Promise.all([fetchUsers()]).then(([users]) => {
```
Comment (Laennart) [inline src/pages/manage/users/activeTrainerTable.tsx:51]:
```suggestion
    fetchUsers().then((users) => {
```

File: src/pages/manage/users/invitedUserTable.tsx [hunk 2]
```diff
@@ -29,20 +52,23 @@ const InvitedUserTable: React.FC<invitedUserTableProps> = ({ ...props }) => {
...
-  const [loading, setLoading] = useState(true);
-  const [error, setError] = useState(false);
   const [data, setData] = useState<User[]>([]);
-  const [, draw] = useState({});
+  const [isMounted, setIsMounted] = useState(true);
 
   const [update, setUpdate] = useState(updateValue);
 
+  const loadData = () => {
+    Promise.all([fetchUsers()]).then(([users]) => {
```
Comment (Laennart) [inline src/pages/manage/users/invitedUserTable.tsx:61]:
```suggestion
    fetchUsers().then((users) => {
```

File: src/pages/train/index.tsx [hunk 3]
```diff
@@ -7,11 +7,27 @@ import { useParams } from "react-router-dom";
...
+  if (!response.success) {
+    message.error(response.description);
+    return [];
+  }
+  return response.data.exercises.find((e: any) => {
+    return e.exercise_plan_id === planId;
+  });
+};
+
+interface TrainProps {
```
Comment (Laennart) [inline src/pages/train/index.tsx:25]:
```suggestion
interface trainProps {
```

File: src/pages/train/index.tsx [hunk 4]
```diff
@@ -7,11 +7,27 @@ import { useParams } from "react-router-dom";
...
+  rawExercise: Record<string, any>;
+}
+
+const Train: React.FC<TrainProps> = ({ rawExercise }) => {
```
Comment (Laennart) [inline src/pages/train/index.tsx:29]:
```suggestion
const Train: React.FC<trainProps> = ({ rawExercise }) => {
```

File: src/pages/train/index.tsx [hunk 5]
```diff
@@ -23,65 +39,42 @@ const Train: React.FC = () => {
...
-        return e.exercise_plan_id === parseInt(exercisePlanId ?? "");
+    if (data && isMounted) {
+      setExercise({
+        title: data.title,
+        description: data.description,
+        sets: rawExercise.sets,
+        repeatsPerSet: rawExercise.repeats_per_set,
+        videoPath:
+          data.video ??
+          "https://vid.pr0gramm.com/2021/12/28/130aaef3ab9c207a.mp4",
```
Comment (Laennart) [inline src/pages/train/index.tsx:55]: I think we also can remove this video 😄

File: src/pages/train/index.tsx [hunk 6]
```diff
@@ -23,65 +39,42 @@ const Train: React.FC = () => {
...
-      // clean up
       isMounted = false;
     };
-  }, [loading, exercisePlanId]);
+  }, [data, rawExercise.repeats_per_set, rawExercise.sets]);
 
   return (
     <Container confimLeave={subPage !== "exerciseDone"}>
-      {loading || exercise === undefined ? (
+      {isLoading || exercise === null ? (
```
Comment (Laennart) [inline src/pages/train/index.tsx:67]: Wouldn't that also work?
```suggestion
      {isLoading || !exercise ? (
```

File: src/pages/train/index.tsx [hunk 7]
```diff
@@ -23,65 +39,42 @@ const Train: React.FC = () => {
...
         <div>Loading...</div>
       ) : (
         <>
           {subPage === "training" && (
             <Training
-              loadingExercise={loading}
+              loadingExercise={false}
```
Comment (Laennart) [inline src/pages/train/index.tsx:73]: If this property is always false you can remove it. (Same with `error` below)

**Review by Laennart — CHANGES_REQUESTED** [review]
Some bugs I noticed:
- I'm not able to create a new plan, the page does not load.
- Progress Bar on Training Page instantly jumps to 100% when starting the first set. On the second set it works somehow
- Countdown on setDone Page does not work

File: src/pages/train/index.tsx (hunk 3 above)
Comment (Gamer92000) [reply src/pages/train/index.tsx:25]: Nope, according to most style guides interfaces in TS should be in PascalCase.
E.g. see https://dirask.com/posts/TypeScript-interface-naming-convention-x1RwqD

Comment (Laennart) [reply src/pages/train/index.tsx:25]: In your previous refactoring PRs you changed everything to lower case 🤔 
E.g. [here](https://github.com/BP-WiSe21-22-Gruppe-52/BP-frontend/blob/07b34fdaf05b1aeb6fcb490fb01c05f018c082ed/src/pages/train/setDone.tsx#L16)

Comment (Gamer92000) [reply src/pages/train/index.tsx:25]: Hmmm 🤔 
Past errors shall not dictate future misconduct.

**Review by Laennart — APPROVED** [review]
🔧

**Pull request merged**